platforms/core-runtime/stdlib-java-extensions/src/main/java/org/gradle/internal/service/scopes/Scope.java

```diff
@@ -17,75 +17,131 @@
 package org.gradle.internal.service.scopes;
 
 /**
+ * Scopes represent the state, services and lifetime of each activity performed by Gradle as part of
+ * executing a <em>Gradle invocation</em> against a <em>user build</em>.
+ * <p>
+ * <b>User build</b> is a collection of files on disk that constitutes the user's software project,
+ * building of which is orchestrated by Gradle.
+ * It can be a Java library, a backend microservice, or an enterprise monorepo.
+ * <p>
+ * <b>Gradle invocation</b> is a direct or indirect request of the user to invoke Gradle
+ * over a given <em>user build</em> to do some work.
+ * It can be a command-line invocation or a Tooling API client request (e.g. an IDE sync).
+ *
  * @see ServiceScope
  */
 public interface Scope {
 
     /**
-     * These services are reused across builds in the same process.
-     *
-     * <p>Global services are visible to all other services.</p>
+     * Scope of the entire <em>build process</em>, e.g. a long-lived Gradle daemon.
+     * <p>
+     * The build process can potentially serve multiple Gradle invocations.
+     * The <em>build process state</em> holds the global state of the build process and manages all the other state.
+     * <p>
+     * The build process state is managed by the {@code BuildProcessState} class.
+     * An instance is created once for a given process.
+     * <p>
+     * Global services are visible to all other services.
      */
     interface Global extends Scope {}
 
     /**
-     * These services are reused across builds in the same process while the Gradle user home directory remains unchanged.
-     * The services are closed when the Gradle user home directory changes.
-     *
-     * <p>{@link Global} and parent scope services are visible to {@link UserHome} scope services, but not vice versa.</p>
+     * The scope of a Gradle invocation particular Gradle user home directory.
+     * <p>
+     * When that directory changes between Gradle invocations, the state is discarded and recreated.
```

Comment:
❔ Can we have multiple `UserHome` scopes when a `GradleBuild` task with custom home dir is used?

Reply:
I don't know, maybe @adammurdoch can clarify